test/join.jl

```diff
@@ -1562,47 +1562,48 @@ end
+        DataFrame(ID=[1, 2, missing],
+                  Name=["John Doe", "Jane Doe", "Joe Blogs"]),
+        DataFrame(ID=Int[],
+                  Name=String[]),
+        DataFrame(ID=Union{Int, Missing}[missings(3)...],
+                  Name=String["John Doe", "Jane Doe", "Joe Blogs"]),
+        DataFrame(ID=[1, 2, 3],
+                  Name=[missing, "Jane Doe", missing]),
+        DataFrame(ID=[1:100; missings(100)],
+                  Name=repeat(["Jane Doe"], 200)),
+        DataFrame(ID=[missings(100); 1:100],
+                  Name=repeat(["Jane Doe"], 200)),
+        DataFrame(ID=[1:50; missings(100); 51:100],
+                  Name=repeat(["Jane Doe"], 200)),
+        DataFrame(ID=[1:64; missings(64); 129:200],
+                  Name=repeat(["Jane Doe"], 200)),
+        DataFrame(ID=[1:63; missings(65); 129:200],
+                  Name=repeat(["Jane Doe"], 200)),
+        DataFrame(ID=rand([1:1000; missing], 10000),
+                  Name=rand(["John Doe", "Jane Doe", "Joe Blogs", missing], 10000)),
     ]
     jobs = [
-        DataFrame(ID = Union{Int, Missing}[1, 2, 2, 4],
-            Job = Union{String, Missing}["Lawyer", "Doctor", "Florist", "Farmer"]),
-        DataFrame(ID = Union{Int, Missing}[missing, 2, 2, 4],
-            Job = Union{String, Missing}["Lawyer", "Doctor", "Florist", "Farmer"]),
-        DataFrame(ID = Union{Int, Missing}[missing, 2, 2, 4],
-            Job = Union{String, Missing}["Lawyer", "Doctor", missing, "Farmer"]),
-        DataFrame(ID = Union{Int, Missing}[],
-            Job = Union{String, Missing}[]),
-        DataFrame(ID = Union{Int, Missing}[1:100; missings(100)],
-            Job = Union{String, Missing}[repeat(["Lawyer"], 200)...]),
-        DataFrame(ID = Union{Int, Missing}[missings(100); 1:100],
-            Job = Union{String, Missing}[repeat(["Lawyer"], 200)...]),
-        DataFrame(ID = Union{Int, Missing}[1:50; missings(100); 51:100],
-            Job = Union{String, Missing}[repeat(["Lawyer"], 200)...]),
-        DataFrame(ID = Union{Int, Missing}[1:64; missings(64); 129:200],
-            Job = Union{String, Missing}[repeat(["Lawyer"], 200)...]),
-        DataFrame(ID = Union{Int, Missing}[1:63; missings(65); 129:200],
-            Job = Union{String, Missing}[repeat(["Lawyer"], 200)...]),
-        DataFrame(ID = Union{Int, Missing}[rand([1:1000; missing], 10000)...],
-            Job = Union{String, Missing}[rand(["Lawyer", "Doctor", "Florist", missing], 10000)...]),
+        DataFrame(ID=[1, 2, 2, 4],
+                  Job=["Lawyer", "Doctor", "Florist", "Farmer"]),
+        DataFrame(ID=[missing, 2, 2, 4],
+                  Job=["Lawyer", "Doctor", "Florist", "Farmer"]),
+        DataFrame(ID=[missing, 2, 2, 4],
+                  Job=["Lawyer", "Doctor", missing, "Farmer"]),
+        DataFrame(ID=Union{Int, Missing}[],
+                  Job=Union{String, Missing}[]),
```

Comment:
```suggestion
        DataFrame(ID=[],
                  Job=[]),
```

Reply:
I wanted to hint that this is empty array of type `Union{Int, Missing}[]`, not `Any[]`.
Same with missings bellow.
Do you think I should remove types all the way?

Reply:
Yes, unless there's a specific reason not to. For example, it would be appropriate if it's needed to cover a particular code path, test a particular case, or when checking that types are equal. But here it's not clear to me that it's the case.

Reply:
Ok, thanks